File: src/main/java/com/procurement/contracting/model/dto/ocds/TenderStatus.java
package com.procurement.contracting.model.dto.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import com.procurement.contracting.exception.EnumException;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public enum TenderStatus {
    PLANNING("planning"),
    PLANNED("planned"),
    ACTIVE("active"),
    CANCELLED("cancelled"),
    UNSUCCESSFUL("unsuccessful"),
    COMPLETE("complete"),
    WITHDRAWN("withdrawn");

    private static final Map<String, TenderStatus> CONSTANTS = new HashMap<>();
    private final String value;

    static {
        for (final TenderStatus c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    TenderStatus(final String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static TenderStatus fromValue(final String value) {
        final TenderStatus constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new EnumException(TenderStatus.class.getName(), value, Arrays.toString(values()));
        }
        return constant;
    }
}

File: src/main/kotlin/com/procurement/contracting/domain/model/tender/status/TenderStatus.kt
package com.procurement.contracting.domain.model.tender.status

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.procurement.contracting.domain.model.EnumElementProvider

enum class TenderStatus(@JsonValue override val key: String) : EnumElementProvider.Key {
    PLANNING("planning"),
    PLANNED("planned"),
    ACTIVE("active"),
    CANCELLED("cancelled"),
    UNSUCCESSFUL("unsuccessful"),
    COMPLETE("complete"),
    WITHDRAWN("withdrawn");

    override fun toString(): String = key

    companion object : EnumElementProvider<TenderStatus>(info = info()) {
        @JvmStatic
        @JsonCreator
        fun creator(name: String) = TenderStatus.orThrow(name)
    }
}
